src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -221,13 +233,39 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
             // Parse first 3 characters after a space as status code
             // TODO: Parsing errors are treated as fatal. Find right behaviour
             int statusCodeLength = 3;
-            int codeIndex = responseHeader.IndexOf(' ') + 1;
-            string strStatusCode = responseHeader.Substring(codeIndex, statusCodeLength);
-            response.StatusCode = (HttpStatusCode)(int.Parse(strStatusCode));
 
-            // The remaining string after the space is the reason phrase
-            response.ReasonPhrase = responseHeader.Substring(codeIndex + statusCodeLength + 1);
+            int spaceIndex = responseHeader.IndexOf(spaceChar);
+
+            if (spaceIndex > -1)
+            {
+                int codeStartIndex = spaceIndex + 1;
+                int codeEndIndex = codeStartIndex + statusCodeLength;
+
+                if (codeEndIndex <= responseHeaderLength)
+                {
+                    string strStatusCode = responseHeader.Substring(codeStartIndex, statusCodeLength);
```

Comment:
There's unfortunately no int.TryParse that accepts a string, offset, and count.  But given how common it is to need parse this status code, I think it'd be worth a bit of custom code to avoid the substring allocation, e.g.
```C#
private static bool TryParseStatusCode(string responseHeader, int offset, out int status)
{
    if (offset + 3 > responseHeader.Length)
    {
        status = 0;
        return false;
    }

    char c100 = responseHeader[offset];
    char c10 = responseHeader[offset + 1];
    char c1 = responseHeader[offset + 2];

    if (c100 < '0' || c100 > '9' ||
        c10 < '0' || c10 > '9' ||
        c1 < '0' || c1 > '9')
    {
        status = 0;
        return false;
    }

    status =
        (c100 - '0') * 100 +
        (c10 - '0') * 10 +
        (c1 - '0');
    return true;
}
```